lib/nx_inflate.c

```diff
@@ -1261,30 +1382,34 @@ static int nx_inflate_(nx_streamp s, int flush)
 			   that is about 2 pages minimum for source and
 			   and 6 pages for target; if the system does not
 			   have 8 free pages then the loop will last forever */
-			source_sz = source_sz - nx_history_len;
-			if (source_sz > (2 * INF_MIN_INPUT_LEN))
-				source_sz = (source_sz + 1) / 2;
-			else if (source_sz > INF_MIN_INPUT_LEN)
-				source_sz = INF_MIN_INPUT_LEN;
-
-			/* else if caller gave fewer source bytes, keep it as is */
-			source_sz = source_sz + nx_history_len;
-
-			if (target_sz > (2 * INF_MAX_EXPANSION_BYTES))
-				target_sz = (target_sz + 1) / 2;
-			else if (target_sz > INF_MAX_EXPANSION_BYTES)
-				target_sz = INF_MAX_EXPANSION_BYTES;
+			source_sz_expected = source_sz - nx_history_len;
+			if (source_sz_expected > (2 * INF_MIN_INPUT_LEN))
+				source_sz_expected = (source_sz_expected + 1)
+						     / 2;
+			else if (source_sz_expected > INF_MIN_INPUT_LEN)
+				source_sz_expected = INF_MIN_INPUT_LEN;
+
+			/* else if caller gave fewer source bytes, keep it as
+			   is.  */
+			source_sz = source_sz_expected + nx_history_len;
+
+			if (target_sz_expected > (2 * INF_MAX_EXPANSION_BYTES))
+				target_sz_expected = (target_sz_expected + 1)
+						     / 2;
```

Comment:
Please put this in the previous line, it's easier to read.

Reply:
Likewise here.